packages/ui-components/src/FormDialog.tsx

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2018-2020 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../style/index.css';
+
+import { Dialog } from '@jupyterlab/apputils';
+import { Widget } from '@lumino/widgets';
+
+const DEFAULT_BUTTON_CLASS = 'elyra-DialogDefaultButton';
+/*
+ * Validate required dialog fields upon display
+ * - Provides a generic validation by checking if required form fields are populated
+ * - Expect required fields in dialog body to contain attribute: data-form-required
+ *
+ * @params
+ *
+ * options - The dialog setup options
+ * formValidationFunction - Optional custom validation function
+ *
+ * returns a call to dialog display
+ */
+export const showFormDialog = async (
+  options: Partial<Dialog.IOptions<any>>,
+  formValidationFunction?: (dialog: Dialog<any>) => void
+): Promise<Dialog.IResult<any>> => {
+  const dialogBody = options.body;
+  const dialog = new Dialog(options);
+
+  if (formValidationFunction) {
+    formValidationFunction(dialog);
+  } else {
+    if (dialogBody instanceof Widget) {
+      const requiredFields: NodeListOf<any> = dialogBody.node.querySelectorAll(
+        '[data-form-required]'
+      );
+
+      if (requiredFields && requiredFields.length > 0) {
+        // Keep track of all fields already validated. Start with an empty set.
+        const fieldsValidated = new Set();
+
+        // Override Dialog.handleEvent to prevent user from bypassing validation upon pressing the 'Enter' key
+        const dialogHandleEvent = dialog.handleEvent;
+        dialog.handleEvent = (event: Event): void => {
+          if (
+            event instanceof KeyboardEvent &&
+            event.type === 'keydown' &&
+            event.keyCode === 13 &&
+            fieldsValidated.size !== requiredFields.length
```

Comment:
Not always since this is a callback for anytime a keyboard button is pressed on the dialog, `fieldsValidated` could already have been populated at some point

Reply:
ah, whoops, this is in the event handler, my bad